### PR TITLE
Update 2.19.3

### PR DESCRIPTION
**Pull request by uwearzt**
Updating the lib to a recent version.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
             features: vendored
             target: x86_64-unknown-linux-gnu
             cache_id: vendored
-          - toolchain: 1.43.0 # MSRV
+          - toolchain: 1.56.0 # MSRV
```
Comment (randombit) [inline .github/workflows/ci.yml:50]: Why is this required?

Comment (uwearzt) [reply .github/workflows/ci.yml:50]: The actual version of ```cstr_core``` does not work with the 1.43 version (Alternative would be pinning to a pre 0.2.6 version of ```cstr_core```).

Comment (randombit) [reply .github/workflows/ci.yml:50]: MSRV only applies for `std` builds, for `no-std` we anyway require nightly.

Comment (randombit) [reply .github/workflows/ci.yml:50]: I think given recent stabilization of `core::ffi` we can just start using that instead on no-std builds.

**Comment by randombit** [comment]
Thanks for the update
